tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+            provider.GetRequiredService<IOptionsFactory<MicrosoftIdentityOptions>>().Create("");
+            configMock.Received(3).GetSection(_configSectionName);
+
+            AddSignIn_TestCommon(services, provider);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void AddSignIn_WithConfigActions(bool useServiceCollectionExtension)
+        {
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+
+            var provider = services.BuildServiceProvider();
+
+            // Assert configure options actions added correctly
+            var configuredOidcOptions = provider.GetServices<IConfigureOptions<OpenIdConnectOptions>>().Cast<ConfigureNamedOptions<OpenIdConnectOptions>>();
+            var configuredMsOptions = provider.GetServices<IConfigureOptions<MicrosoftIdentityOptions>>().Cast<ConfigureNamedOptions<MicrosoftIdentityOptions>>();
+
+            Assert.Contains(configuredOidcOptions, o => o.Action == _configureOidcOptions);
+            Assert.Contains(configuredMsOptions, o => o.Action == _configureMsOptions);
+
+            AddSignIn_TestCommon(services, provider);
+        }
+
+        private void AddSignIn_TestCommon(IServiceCollection services, ServiceProvider provider)
+        {
+            // Assert correct services added           
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<OpenIdConnectOptions>));
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<MicrosoftIdentityOptions>));
+            Assert.Contains(services, s => s.ServiceType == typeof(IOpenIdConnectMiddlewareDiagnostics));
+            Assert.Equal(ServiceLifetime.Singleton, services.First(s => s.ServiceType == typeof(IOpenIdConnectMiddlewareDiagnostics)).Lifetime);
+            Assert.Contains(services, s => s.ServiceType == typeof(IPostConfigureOptions<CookieAuthenticationOptions>));
+
+            // Assert OIDC options added correctly
+            var configuredOidcOptions = provider.GetService<IConfigureOptions<OpenIdConnectOptions>>() as ConfigureNamedOptions<OpenIdConnectOptions>;
+
+            Assert.Equal(_oidcScheme, configuredOidcOptions.Name);
+
+            // Assert properties set
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            Assert.Equal(_cookieScheme, oidcOptions.SignInScheme);
+            Assert.NotNull(oidcOptions.Authority);
+            Assert.NotNull(oidcOptions.TokenValidationParameters.IssuerValidator);
```

Comment:
Is there a reason to not check the values here, as you do for preferredUserName? Might as well check the authority is correct, client_id, callback path, scopes, issuer address, etc...

Reply:
I was only testing the specific things to that method. Valid authority format is tested in AuthorityHelpers and other properties aren't touched here.